docs/writing-declarations.md

```diff
@@ -2,19 +2,19 @@
 title: Writing Declarations
```

Comment:
Please also undo the file move and the sidebars change

Reply:
Sorry, sidebar is fine, docusaurus config should be reverted if this file moves back I guess?

Reply:
i dont understand the question.

to clarify, in this PR i moved "writing declarations" from advanced --> normal, since I felt it was more appropriate now that it also acts as a mini-tutorial on what type declarations actually are.

Reply:
Like I said in the original comment, I think this will change the url this is on, breaking any links that are out there. The page is already not in advanced anymore in the sidebar, file location doesn't really matter too much.

Reply:
none of the links are broken, as the site wouldn't build and CI wouldn't pass otherwise

Reply:
I'm talking about links from the outside to the docs like https://typescripttolua.github.io/docs/advanced/writing-declarations would break if you moved the file, so I'd rather keep it in the same place

Reply:
oh, i see, okay